### ISSUE TITLE
Problem with creating forms with long names (database table problem)

### ISSUE THREAD

**Issue by ragnarfrosti** [verbb/formie#759]
One of our clients likes to create forms with really long names, and this is causing some issues.

We just updated to Formie version (1.5.9) which did not help.

If I try to create a form with a long name: 
**This is a test form with a very long name to test the database table functionality** 

then Formie reduces the handle to:
**thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunctional**

But when I click save, the form can't be saved:

![Formie](https://user-images.githubusercontent.com/4622136/154273513-55939097-0d58-4ee8-98f3-9f64f9c8d4f3.png)

And in the Formie log I get the following error:
`2022-02-16 14:22:41 [-][2918][-][error][formie] Unable to save form “thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunctional”: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'fieldContentTable' at row 1`

If I add one more letter to the handle I get the following error:
**Handle should contain at most 64 characters**

So I tried to remove the last letter from the handle and save the form until it finally worked:

- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunctional** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunctiona** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunction** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunctio** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFuncti** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunct** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFunc** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFun** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableFu** [FAIL]
- **thisIsATestFormWithAVeryLongNameToTestTheDatabaseTableF** [SUCCESS]

And when I looked into the database then Formie has created the tables for the last 5 fails:

![Formie-tables](https://user-images.githubusercontent.com/4622136/154275686-1194e46a-44cd-4a1f-a3e9-ae17a97a5d7a.png)

So in my case the table name can't be longer than 56 characters + fmc_ = 60 characters.

I would really appreciate if the handle was reduced to fewer characters to prevent this error and to get a better error message for the client (that the handle is to long as the 64 characters limit does not work in my case). And it would also be nice if the tables would not be created on error :)

**Additional info**
- Plugin version: 1.5.9
- Craft version: 3.7.25.1

**Comment by engram-design** [comment]
That's interesting that the tables _are_ created on error, and it also should be truncating the handle. I'll take another look.

**Comment by engram-design** [comment]
Ah, I know the issue - related to the `fieldContentTable` column for a form which is smaller than it should be. Hence the content tables being created, but an error saving the resulting content table to the form. Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.9",
  "...": "..."
}
```

Then run `composer update`.

**Comment by ragnarfrosti** [comment]
Thanks for looking into this. I'll try to have a look at the fix later today :)

**Comment by engram-design** [comment]
Fixed in [1.5.10](https://github.com/verbb/formie/releases/tag/1.5.10)

**Comment by ragnarfrosti** [comment]
Hi @engram-design 

I just updated Formie to version 1.5.10 and now when I create a Formie form with the name **This is a new test form with a very long name to test the database table functionality** the form gets created, but the table is not.

The automatic handle generated to the form by Formie is **thisIsANewTestFormWithAVeryLongNameToTestTheDatabaseTableFunctio**. 

So when I try to add fields to the form and save the form I get the following error:
```
2022-02-28 11:15:59 [-][2918][-][error][formie] Unable to save form “thisIsANewTestFormWithAVeryLongNameToTestTheDatabaseTableFunctio”: SQLSTATE[42000]: Syntax error or access violation: 1103 Incorrect table name 'fmc_thisisanewtestformwithaverylongnametotestthedatabasetablefunctio'
The SQL being executed was: SHOW FULL COLUMNS FROM `fmc_thisisanewtestformwithaverylongnametotestthedatabasetablefunctio`
```

I need to change the handle to **thisIsANewTestFormWithAVeryLongNameToTestTheDatabaseTableFun** (remove the last 4 characters) to get the form to start working.

I'm running Nitro 2 with MariaDB 10.5 if that helps

**Comment by engram-design** [comment]
Yup, sorry about this. Properly fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.5.10",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.5.11](https://github.com/verbb/formie/releases/tag/1.5.11)
